cmd/acr/purge.go

```diff
@@ -400,27 +425,30 @@ func dryRunPurge(ctx context.Context, acrClient api.AcrCLIClientInterface, login
 		return -1, -1, err
 	}
 	lastTag := ""
-	tagsToDelete, lastTag, err := getTagsToDelete(ctx, acrClient, repoName, regex, timeToCompare, "")
-	if err != nil {
-		return -1, -1, err
-	}
 	// The loop to get the deleted tags follows the same logic as the one in the purgeTags function
-	for len(lastTag) > 0 {
-		for _, tag := range *tagsToDelete {
-			// For every tag that would be deleted first check if it exists in the map, if it doesn't add a new key
-			// with value 1 and if it does just add 1 to the existent value.
-			if _, exists := deletedTags[*tag.Digest]; exists {
-				deletedTags[*tag.Digest]++
-			} else {
-				deletedTags[*tag.Digest] = 1
-			}
-			fmt.Printf("%s/%s:%s\n", loginURL, repoName, *tag.Name)
-			deletedTagsCount++
-		}
-		tagsToDelete, lastTag, err = getTagsToDelete(ctx, acrClient, repoName, regex, timeToCompare, lastTag)
+	for {
+		tagsToDelete, newLastTag, err := getTagsToDelete(ctx, acrClient, repoName, regex, timeToCompare, lastTag)
+		lastTag = newLastTag
```

Comment:
Can we check the error first? Same as we did in purgeTags()
https://github.com/Azure/acr-cli/pull/73/files#diff-6a5ebae231898bd993422487862b3818R179

Reply:
Missed this one 🤦